packages/appsync-modelgen-plugin/src/visitors/appsync-visitor.ts

```diff
@@ -407,6 +446,32 @@ export class AppSyncModelVisitor<
       values,
     };
   }
+
+  UnionTypeDefinition(node: UnionTypeDefinitionNode): void {
+    if (this.typesToSkip.includes(node.name.value)) {
+      return;
+    }
+    const unionObject: CodeGenUnion = {
+      name: node.name.value,
+      type: 'union',
+      typeNames: node.types?.map(type => type.name.value) ?? [],
+    }
+    this.unionMap[node.name.value] = unionObject;
+  }
+
+  InterfaceTypeDefinition(node: InterfaceTypeDefinitionNode): void {
+    if (this.typesToSkip.includes(node.name.value)) {
+      return;
+    }
+    const fields = (node.fields as unknown) as CodeGenField[];
```

Comment:
same q about type casting.

Reply:
As mentioned above, the fields are casted by `FieldTypeDefinition`, which is executed ahead of this function.